File: Go/Problem_69.go
package main

import (
	"fmt"
)

func mySqrt(x int) int {
	if x < 0 {
		panic("Negetive numbers not allowed")
	} else {
		start := 0
		end := x
		ans := -1

		for start <= end {
			mid := (start + end) / 2
			square := mid * mid

			if square == x {
				return mid
			} else if square < x {
				ans = mid
				start = mid + 1
			} else {
				end = mid - 1
			}
		}

		return ans

	}
}

func execute_problem69(value int) int {

	fmt.Printf("Root: %d\n", mySqrt(value))

	return 0
}

File: Problem_69.java
public class Problem_69 {

    public static int mySqrt(int x) {
        if(x<0){
            throw new IllegalArgumentException("Negetive numbers not allowed");
        } else {
            int start = 0, end = x, ans = -1;

            while(start <= end){
                int mid = (start + end) / 2;
                int square = mid * mid;

                if(square == x){
                    return square;
                }
                else if(square < x){
                    ans = mid;
                    start = mid + 1;
                } else {
                    end = mid - 1;
                }
            }

            return ans;
        }
    }

    public static void main(String[] args) {
        int answer = mySqrt(10);

        System.out.println("Result: " + answer);
    }
}